test/functional/helpers.lua

```diff
@@ -566,23 +584,84 @@ function module.nvim_async(method, ...)
   session:notify('nvim_'..method, ...)
 end
 
-function module.buffer(method, ...)
-  return module.request('nvim_buf_'..method, ...)
-end
+module.async_meths = module.create_callindex(module.nvim_async)
+module.uimeths = module.create_callindex(ui)
 
-function module.window(method, ...)
-  return module.request('nvim_win_'..method, ...)
-end
+local function create_api(request, call)
+  local m = {}
+  function m.nvim(method, ...)
+    return request('nvim_'..method, ...)
+  end
+
+  function m.buffer(method, ...)
+    return request('nvim_buf_'..method, ...)
+  end
+
+  function m.window(method, ...)
+    return request('nvim_win_'..method, ...)
+  end
+
+  function m.tabpage(method, ...)
+    return request('nvim_tabpage_'..method, ...)
+  end
+
+  function m.curbuf(method, ...)
+    if not method then
+      return m.nvim('get_current_buf')
+    end
+    return m.buffer(method, 0, ...)
+  end
 
-function module.tabpage(method, ...)
-  return module.request('nvim_tabpage_'..method, ...)
+  function m.curwin(method, ...)
+    if not method then
+      return m.nvim('get_current_win')
+    end
+    return m.window(method, 0, ...)
+  end
+
+  function m.curtab(method, ...)
+    if not method then
+      return m.nvim('get_current_tabpage')
+    end
+    return m.tabpage(method, 0, ...)
+  end
+
+  m.funcs = module.create_callindex(call)
+  m.meths = module.create_callindex(m.nvim)
+  m.bufmeths = module.create_callindex(m.buffer)
+  m.winmeths = module.create_callindex(m.window)
+  m.tabmeths = module.create_callindex(m.tabpage)
+  m.curbufmeths = module.create_callindex(m.curbuf)
+  m.curwinmeths = module.create_callindex(m.curwin)
+  m.curtabmeths = module.create_callindex(m.curtab)
+
+  return m
 end
 
-function module.curbuf(method, ...)
-  if not method then
-    return module.nvim('get_current_buf')
+module.rpc = {
+  api = create_api(module.request, module.call),
+}
+
+module.lua = {
+  api = create_api(module.request_lua, module.call_lua),
+}
+
+module.describe_lua_and_rpc = function(describe)
+  return function(what, tests)
+    local function d(flavour)
+      describe(string.format('%s (%s)', what, flavour), function(...)
+        return tests(module[flavour].api, ...)
+      end)
+    end
+
+    d('rpc')
+    d('lua')
   end
-  return module.buffer(method, 0, ...)
+end
+
+-- TODO(lewis6991): Remove this ad ref helpers.api.rpc directly
```

Comment:
I don't think this should be removed. RPC is how the test runner is accessing the neovim API.